### PR TITLE
Fix/request renderer behaviour

### PR DESCRIPTION
**Pull request by Vin-RM**
Fix DecidableReadAttributeRequest's behaviour.

### CODE REVIEW

**Review by nicole-eb — APPROVED** [review]

File: packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart [hunk 1]
```diff
@@ -24,34 +23,53 @@ class ValueRendererListTile extends StatelessWidget {
...
+
   @override
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.symmetric(vertical: 8),
       child: Row(
         children: [
-          if (checkboxSettings != null) Checkbox(value: checkboxSettings!.isChecked, onChanged: checkboxSettings!.onUpdateCheckbox),
+          if (widget.checkboxSettings != null)
+            Checkbox(value: widget.checkboxSettings!.isChecked, onChanged: widget.checkboxSettings!.onUpdateCheckbox),
```
Comment (jkoenig134) [inline packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart:61]: missing trailing comma

File: packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart [hunk 2]
```diff
@@ -24,34 +23,53 @@ class ValueRendererListTile extends StatelessWidget {
...
+
+      widget.onUpdateInput(
+        inputValue: inputValue,
+        valueType: widget.valueType,
+        isComplex: widget.renderHints.editType == RenderHintsEditType.Complex ? true : false,
+      );
+    });
+  }
+
   @override
```
Comment (jkoenig134) [inline packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart:54]: you forgot to dispose the controller

File: packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart [hunk 3]
```diff
@@ -24,34 +23,53 @@ class ValueRendererListTile extends StatelessWidget {
...
+    required this.onUpdateInput,
   });
 
+  @override
+  State<ValueRendererListTile> createState() => _ValueRendererListTileState();
+}
+
+class _ValueRendererListTileState extends State<ValueRendererListTile> {
+  final ValueRendererController controller = ValueRendererController();
+  dynamic inputValue;
```
Comment (jkoenig134) [inline packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart:37]: remove

File: packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart [hunk 4]
```diff
@@ -24,34 +23,53 @@ class ValueRendererListTile extends StatelessWidget {
...
+
+  @override
+  void initState() {
+    super.initState();
+
+    controller.addListener(() {
+      setState(() => inputValue = controller.value);
```
Comment (jkoenig134) [inline packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart:44]: remove

File: packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart [hunk 5]
```diff
@@ -24,34 +23,53 @@ class ValueRendererListTile extends StatelessWidget {
...
+
+      widget.onUpdateInput(
+        inputValue: inputValue,
```
Comment (jkoenig134) [inline packages/renderers/lib/src/renderers/widgets/value_renderer_list_tile.dart:47]:
```suggestion
        inputValue: controller.value,
```

File: packages/renderers/lib/src/renderers/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart [hunk 6]
```diff
@@ -100,17 +100,60 @@ class _DecidableReadAttributeRequestItemRendererState extends State<DecidableRea
     );
   }
 
+  void onUpdateInput({String? valueType, dynamic inputValue, required bool isComplex}) {
```
Comment (jkoenig134) [inline packages/renderers/lib/src/renderers/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart:103]: don't we need this new logic for propose?

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: packages/renderers/lib/src/renderers/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart (hunk 6 above)
Comment (Vin-RM) [reply packages/renderers/lib/src/renderers/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart:103]: not applicable

**Review by jkoenig134 — APPROVED** [review]

**Review by jkoenig134 — DISMISSED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**